src/Carnac.Logic/KeysController.cs

```diff
@@ -20,8 +22,12 @@ public KeysController(ObservableCollection<Message> messages, IMessageProvider m
             this.messages = messages;
             this.messageProvider = messageProvider;
             this.concurrencyService = concurrencyService;
-        }
 
+            var settingsProvider = new SettingsProvider(new RoamingAppDataStorage("Carnac"));
```

Comment:
@shiftkey I spent some time searching for an existing provider, but it seems like the scope for the existing ones are limited, and the only other solution is passing a reference in what seemed like a complicated way. 

I will look into this more closely this week and try to find a better solution.